src/main/notifications/index.js

```diff
@@ -24,9 +26,20 @@ export function displayMention(title, body, channel, teamId, silent, webcontents
         silent,
         data,
     };
-    const mention = new Mention(options);
+
+    const mention = new Mention(options, channel);
 
     mention.on('show', () => {
+        // On Windows, manually dismiss notifications from the same channel and only show the latest one
+        if (process.platform === 'win32') {
+            for (const notification of currentNotifications.values()) {
```

Comment:
Instead of using a `Set` and iterating over it, you could use a `Map` mapping channel IDs to pending notifications instead. 0/5 on that though

Reply:
I agree with @hmhealey using a composite key in a map makes more sense, as we want to have a unique element for the key, but the value will be different.